plugins/guests/linux/cap/public_key.rb

```diff
@@ -53,6 +53,7 @@ def self.remove_public_key(machine, contents)
               result=$?
             fi
             rm -f '#{remote_path}'
+            chmod 0600 ~/.ssh/authorized_keys
```

Comment:
Hmm, it looks to still be there on the master branch, as far as I can tell. Check it out [here](https://github.com/mitchellh/vagrant/blob/a6760dd8e7743e048cb2f38c474e05889356e8ac/plugins/guests/linux/cap/public_key.rb#L57)